frappe/utils/user.py

```diff
@@ -66,8 +69,14 @@ def build_doctype_map(self):
 		active_domains = frappe.get_active_domains()
 
 		self.doctype_map = {}
-		for r in frappe.db.sql("""select name, in_create, issingle, istable,
-			read_only, restrict_to_domain, module from tabDocType""", as_dict=1):
+
+		table = builder.get_reflection('tabDocType')
```

Comment:
Creating table schema from DB at runtime can make the system very slow.

Reply:
One alternative to this would be generate reflections for all tables for all sites and cache them. Then pick out the reflection from cache each time. 

Can this be done differently?